### PR TITLE
Fix/584 add force download params in file endpoints +tests

### PR DESCRIPTION
**Pull request by inkhey**
related to tracim/tracim#854

### CODE REVIEW

File: backend/tracim_backend/models/context_models.py [hunk 1]
```diff
@@ -205,14 +205,27 @@ def __init__(self, acp: str):
         self.acp = acp
 
 
+class FileQuery(object):
```
Comment (lebouquetin) [inline backend/tracim_backend/models/context_models.py]: What is the point of creating `FileQuery` class?

File: backend/tracim_backend/views/contents_api/file_controller.py [hunk 2]
```diff
@@ -200,7 +219,11 @@ def preview_pdf_full(self, context, request: TracimRequest, hapic_data=None):  #
             content_type=CONTENT_TYPES.Any_SLUG
         )
         pdf_preview_path = api.get_full_pdf_preview_path(content.revision_id)
-        return FileResponse(pdf_preview_path)
+        response = FileResponse(pdf_preview_path)
+        # INFO - G.M - 2018-09-05 - Support for force download param
+        if hapic_data.query.force_download:
+            response.content_disposition = 'attachment'
```
Comment (lebouquetin) [inline backend/tracim_backend/views/contents_api/file_controller.py]: @inkhey you must factorize this kind of code!

Two ways of doing this:
- add a parameter force_download in `FileResponse` class
- implement and use a decorator

You must factorize this kind of code

File: backend/tracim_backend/models/context_models.py (hunk 1 above)
Comment (inkhey) [reply backend/tracim_backend/models/context_models.py]: Allow turning marshmallow schema into object, Like most object in  backend/tracim_backend/views/core_api/schemas.py

File: backend/tracim_backend/views/contents_api/file_controller.py (hunk 2 above)
Comment (inkhey) [reply backend/tracim_backend/views/contents_api/file_controller.py]: done.

**Pull request merged**